app/models/school.rb

```diff
@@ -18,6 +18,10 @@ class School < ApplicationRecord
   validates :country_code, presence: true, inclusion: { in: ISO3166::Country.codes }
   validates :reference, uniqueness: { case_sensitive: false, allow_nil: true }, presence: false
   validates :district_nces_id, uniqueness: { case_sensitive: false, allow_nil: true }, presence: false
+  validates :school_roll_number,
+            uniqueness: { conditions: -> { where(rejected_at: nil) }, case_sensitive: false, allow_nil: true },
+            presence: { if: :ireland? },
```

Comment:
I think it would be a little more idiomatic and transparent to just stick with 

`presence: { if: ->(school) { school.country_code == 'IE' } },`

it feels overkill to add a custom method for a single usage like this